sale_channel_mirakl/tests/test_products_and_offers_exporter.py

```diff
@@ -0,0 +1,281 @@
+            request_type=None,
+        ):
+            self.url = url
+            self.headers = headers
+            self.files = files
+            self.request_type = request_type
+
+        sale_channel._patch_method("_process_request", _mock_process_request)
+        yield
+        sale_channel._revert_method("_process_request")
+
+    def test_post_products_file_on_mirakl(self):
+        expected_filename = "{}_{}".format(
+            "Product", self.mirakl_sc_for_product.offer_filename
+        )
+        expected_file_content = (
+            '"sku";"ean";"PRODUCT_TITLE";"PRODUCT_DESCRIPTION";"PRODUCT_CAT_CODE"\r\n'
+            '"{p2_dfl}";"{p2_brcd}";"{p2_name}";"{p2_desc}";"{cat}"{car_return}'
+            '"{p1_dfl}";"{empty}";"{p1_name}";"{p1_name}";"{cat}"{car_return}'.format(
+                p2_dfl=self.product2.default_code,
+                p2_brcd=self.product2.barcode,
+                p2_name=self.product2.name,
+                p2_desc=self.product2.description,
+                cat=CAT_CODE,
+                car_return=CARRIAGE_RETURN,
+                p1_dfl=self.product1.default_code,
+                empty=EMPTY_STRING,
+                p1_name=self.product1.name,
+            )
+        )
+
+        with self._patch_process_request(self.mirakl_sc_for_product):
+            self.mirakl_sc_for_product.channel_id._scheduler_export()
+            self._check_parameters_test(
+                EXPECTED_PRODUCT_URL,
+                {"file": (expected_filename, expected_file_content)},
+                REQUEST_POST,
+            )
+
+    def test_post_offers_file_on_mirakl(self):
+        expected_filename = "{}_{}".format(
+            "Offer", self.mirakl_sc_for_offer.offer_filename
+        )
+        expected_file_content = (
+            '"sku";"product-id";"product-id-type";"state"\r\n'
+            '"{p2_dfl}";"{p2_brcd}";"{ean}";"{state}"{car_return}'
+            '"{p1_dfl}";"{p1_dfl}";"{prod_id_type}";"{state}"{car_return}'.format(
+                p2_dfl=self.product2.default_code,
+                p2_brcd=self.product2.barcode,
+                ean=EAN,
+                state=STATE,
+                car_return=CARRIAGE_RETURN,
+                p1_dfl=self.product1.default_code,
+                prod_id_type=PROD_ID_TYPE,
+            )
+        )
+
+        with self._patch_process_request(self.mirakl_sc_for_offer):
+            self.mirakl_sc_for_offer.channel_id._scheduler_export()
+            self._check_parameters_test(
+                EXPECTED_OFFER_URL,
+                {
+                    "file": (expected_filename, expected_file_content),
+                    "import_mode": "NORMAL",
+                },
+                REQUEST_POST,
+            )
+
+    def test_post_catalog_file_on_mirakl(self):
+        expected_filename = self.mirakl_sc_for_offer.offer_filename
+        expected_file_content = (
+            '"sku";"ean";"PRODUCT_TITLE";"PRODUCT_DESCRIPTION";"PRODUCT_CAT_CODE"'
```

Comment:
These headers seems to be the same (depending on the file to export) and should be into a dedicated variable